xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateVersioningStore.java

```diff
@@ -288,39 +301,44 @@ public void updateXWikiDocArchive(XWikiDocument doc, boolean bTransaction, XWiki
     }
 
     /**
+     * Loads all the RCS nodes present in the archive of a given document.
+     *
      * @param context the XWiki context
      * @param id {@link XWikiRCSNodeContent#getId()}
      * @param bTransaction should store to use old transaction(false) or create new (true)
-     * @return loaded rcs node content
+     * @return loaded RCS nodes content
      * @throws XWikiException if any error
      */
     protected List<XWikiRCSNodeInfo> loadAllRCSNodeInfo(XWikiContext context, final long id, boolean bTransaction)
         throws XWikiException
+    {
+        return loadRCSNodeInfo(context, id, new RevisionCriteria(null, null, null, true));
+    }
+
+    /**
+     * Loads a part of the RCS nodes present in the archive of a given document, based on criteria.
+     *
+     * @param context the XWiki context
+     * @param id {@link XWikiRCSNodeContent#getId()}
+     * @param criteria the criteria matching the nodes to load
+     * @return loaded RCS nodes content
+     * @throws XWikiException if any error
+     */
+    protected List<XWikiRCSNodeInfo> loadRCSNodeInfo(XWikiContext context, final long id, RevisionCriteria criteria)
```

Comment:
Missing since

Reply:
Not sure if we put unstable on protected methods of public APIs ?

Reply:
Anything no private is a public API so yes missing `@Unstable` too.